metrics/web/lib/common/presentation/decoration/bubble_shape_border.dart

```diff
@@ -0,0 +1,154 @@
+import 'dart:math';
+import 'package:flutter/material.dart';
+
+/// Enum that represents the side of the shape to display the arrow on.
+enum BubblePosition { bottom, top, left, right }
+
+/// Enum that represents the alignment of the arrow.
+enum BubbleAlignment { start, center, end }
+
+/// A shape border with the arrow.
+class BubbleShapeBorder extends ShapeBorder {
+  /// A side of the shape to display the arrow on.
+  final BubblePosition position;
+
+  /// An alignment of the arrow.
+  final BubbleAlignment alignment;
+
+  /// A size of the arrow.
+  final Size arrowSize;
+
+  /// An offset of the arrow.
+  /// Sets the shift horizontally or vertically depending on the [position].
+  /// Can be a negative.
+  final double offset;
+
+  /// A radius of the border of this shape.
+  final double borderRadius;
+
+  /// Creates a new instance of the [BubbleShapeBorder].
+  ///
+  /// The [position] defaults to the [BubblePosition.bottom].
+  /// The [alignment] defaults to the [BubbleAlignment.center].
+  /// The [arrowSize] defaults to the [Size.square] of 10.0.
+  /// The [offset] default value is 0.0.
+  /// The [borderRadius] default value is 12.0.
+  const BubbleShapeBorder({
+    this.position = BubblePosition.bottom,
+    this.alignment = BubbleAlignment.center,
+    this.arrowSize = const Size.square(10.0),
+    this.offset = 0.0,
+    this.borderRadius = 12.0,
+  });
+
+  @override
+  EdgeInsetsGeometry get dimensions => EdgeInsets.zero;
+
+  /// Indicates whether this arrow on the vertical axis.
+  bool get _isVertical =>
+      position == BubblePosition.top || position == BubblePosition.bottom;
+
+  @override
+  Path getInnerPath(Rect rect, {TextDirection textDirection}) {
+    return null;
+  }
+
+  @override
+  Path getOuterPath(Rect rect, {TextDirection textDirection}) {
+    final path = Path();
+
+    final topLeftDiameter = max(borderRadius, 0);
+    final topRightDiameter = max(borderRadius, 0);
+    final bottomLeftDiameter = max(borderRadius, 0);
+    final bottomRightDiameter = max(borderRadius, 0);
+
+    final spacingLeft = position == BubblePosition.left ? arrowSize.width : 0.0;
+    final spacingTop = position == BubblePosition.top ? arrowSize.height : 0.0;
+    final spacingRight =
+        position == BubblePosition.right ? arrowSize.width : 0.0;
+    final spacingBottom =
+        position == BubblePosition.bottom ? arrowSize.height : 0.0;
+
+    final double left = spacingLeft + rect.left;
```

Comment:
Can we remove the type annotation to be more consistent? 
```suggestion
    final left = spacingLeft + rect.left;
```